.script/tests/detectionTemplateStructureValidation/Models/ValidEntityMappingsAttribute.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.Sentinel.Analytics.Management.AnalyticsManagement.Contracts.Model.ARM.ModelValidation
+{
+    public class ValidEntityMappingsAttribute : ValidationAttribute
+    {
+        private readonly int _entityMappingsMinLength;
+        private readonly int _entityMappingsMaxLength;
+        private readonly int _fieldMappingsMinLength;
+        private readonly int _fieldMappingsMaxLength;
+        private readonly int _laColumnNameMaxLength = 500; // 500 is the max length of a column name in LA
+
+        // field mapping column name should start with a letter or an underscore and contain only alphanumeric English characters (i.e. [a-zA-Z0-9_])
+        private readonly Regex _laColumnNameRegex = new Regex("^[a-zA-Z_]+\\w*$");
+
+        public ValidEntityMappingsAttribute(int entityMappingsMinLength, int entityMappingsMaxLength, int fieldMappingsMinLength, int fieldMappingsMaxLength)
+        {
+            _entityMappingsMinLength = entityMappingsMinLength;
+            _entityMappingsMaxLength = entityMappingsMaxLength;
+            _fieldMappingsMinLength = fieldMappingsMinLength;
+            _fieldMappingsMaxLength = fieldMappingsMaxLength;
+        }
+
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            if (value == null)
+            {
+                return ValidationResult.Success;
+            }
+
+            var entityMappings = (List<EntityMapping>)value;
+            var fieldName = validationContext.MemberName;
```

Comment:
should you validate that validationContext is not null?

Reply:
it won't be. it gets called by the runtime and it will always be passed